care/facility/api/viewsets/facility_users.py

```diff
@@ -0,0 +1,35 @@
+from django_filters import rest_framework as filters
+from rest_framework import mixins
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.viewsets import GenericViewSet
+
+from care.facility.models.facility import Facility
+from care.users.api.serializers.user import UserAssignedSerializer
+from care.users.models import User
+
+
+class UserFilter(filters.FilterSet):
+    user_type = filters.TypedChoiceFilter(choices=[(key, key) for key in User.TYPE_VALUE_MAP.keys()],
+                                          coerce=lambda role: User.TYPE_VALUE_MAP[role])
+
+    class Meta:
+        model = User
+        fields = []
+
+
+class FacilityUserViewSet(GenericViewSet, mixins.ListModelMixin):
+    lookup_field = "external_id"
```

Comment:
leave this if you are not using the retrieve mixin

Reply:
updated it